### PR TITLE
added wilty dup, unc31 classified as complete, warn on missing neuron…

### PR DESCRIPTION
**Pull request by bdsinger**
… when complete animal not being displayed

### CODE REVIEW

**Review by lparsons — COMMENTED** [review]
This looks good to me, but I would like to test it out. Can you paste in the steps I need to rebuild the db?

Also, I don't think we want to add the `environment.yml` file, at least not one with the `prefix` specified.

File: test/model-tests/data-service.json [hunk 1]
```diff
@@ -10633,8 +10633,32 @@
         "ALN",
         "ALNR",
         "ALN",
+        "ANALDEP",
+        "DEFECATIONMUSCLES",
```
Comment (lparsons) [inline test/model-tests/data-service.json:10637]: I'm not clear why this file changed so much.

Comment (lparsons) [reply test/model-tests/data-service.json:10637]: > I'm not clear why this file changed so much.

Ah, I think this has to do with the unc-31 dataset being "complete animal" now?

File: environment.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  - libcxx=12.0.0=h2f01273_0
+  - libdb=6.2.32=h0a44026_0
+  - libedit=3.1.20210910=hca72f7f_0
+  - libuv=1.40.0=haf1e3a3_0
+  - mysql=5.7.24=h56848d4_0
+  - ncurses=6.3=hca72f7f_2
+  - nodejs=14.8.0=hec0b97e_0
+  - openssl=1.1.1m=hca72f7f_0
+  - zlib=1.2.11=h4dc903c_4
+prefix: /Users/bdsinger/miniconda3/envs/funconn
```
Comment (hepcat72) [inline environment.yml:18]: Q. What does this file do? I'm pretty ignorant of yaml file usages in general. Should this be stuff that's added to the `package.json` file?

Comment (hepcat72) [inline environment.yml:18]: B. This hard-coded absolute path (`/Users/bdsinger/miniconda3/envs/funconn`) shouldn't be in the repo. Maybe this file should be in `.gitignore`?

File: src/client/js/model/input.js [hunk 3]
```diff
@@ -136,7 +136,8 @@ ModelPrototype.setInputFromUrlState = function(rawInput) {
           (allInputsNotInDataset.length > 1 ? 'are' : 'is') +
           ' not in the ' +
           this.database +
-          ' datasets.',
+          ' datasets.' +
+          ' Please select complete animal to view this neuron.',
```
Comment (hepcat72) [inline src/client/js/model/input.js:140]: C. Nice touch. Does this render OK? It should show up in a translucent red box if I'm not mistaken. How can I trigger it so I can view it?

**Review by hepcat72 — CHANGES_REQUESTED** [review]
I haven't tested yet. I have a few questions, but the main thing is the hard-coded absolute path, though I can't say I know what that yaml file is for.

File: src/client/js/model/input.js (hunk 3 above)
Comment (lparsons) [reply src/client/js/model/input.js:140]: There are a few example neurons in the issue (#105) that trigger this (`M3`, `MI`, `I3`, etc.)

File: test/model-tests/data-service.json (hunk 1 above)
Comment (bdsinger) [reply test/model-tests/data-service.json:10637]: yes that's it

**Comment by bdsinger** [comment]
image

File: docs/understanding-funconn-data.md [hunk 4]
```diff
@@ -9,7 +9,7 @@ A dataset is associated with connections data, e.g. connections A,B, C are found
 
 - ```name```: name of the dataset
 - ```id```: id of the dataset
-- ```type```: dataset collection. one of 'head', 'tail', 'complete', or 'unc31' (mutant head data)
+- ```type```: dataset collection. one of 'head', 'tail', 'complete', or 'unc31' (mutant data)
```
[the remark this answers is not in the document]
Comment (bdsinger) [reply docs/understanding-funconn-data.md]: thanks for pointing that out, made the change by adding "complete" where "head" was before

File: environment.yml (hunk 2 above)
Comment (bdsinger) [reply environment.yml]: mistake, removed that file

File: test/model-tests/data-service.json (hunk 1 above)
Comment (bdsinger) [reply test/model-tests/data-service.json:10637]: btw to rebuild db can "npm run populate-database" and "npm run build-prod" which I think includes the former command as part of the things it does, but to be sure you'd want to do both.

Comment (lparsons) [reply test/model-tests/data-service.json:10637]: Thx. I don't think you need to do `npm run build-prod` if you're only hosting it via `npm run watch`, but I'm not 100% sure.

**Review by lparsons — APPROVED** [review]
LGTM! :shipit:

File: test/model-tests/data-service.json (hunk 1 above)
Comment (hepcat72) [reply test/model-tests/data-service.json:10637]: I was rebuilding the db (only when necessary) by doing:

```
sh scripts/setup_config.sh "$DBNAME" "$DBUSER" "$PASSWD" | sudo mysql
```

but that should only be necessary if the database schema changed.

**Review by hepcat72 — APPROVED** [review]
Great job.

**Pull request merged**